docs/kns_apps_tutorial.md

```diff
@@ -0,0 +1,347 @@
+                    {
+                        "name": "google-sample",
+                        "type": "kubectl",
+                        "path": "google-sample.yml"
+                    }
+                ]
+            }
+```
+
+`nestedkubernetes` is the new key that allows to define inside it all needed elements for a Kubernetes based app.
+Lets see each parts of this new dict.
+
+### 2.1. General parameters
+
+```json
+            "nestedkubernetes": {
+                "kubernetes_version": "",
+```
+
+* `kubernetes_version` allows to specify the Kubernetes version to be used for the K3S nested cluster. Format is `v1.X.Y` with `X` the major and `Y` the minor versions. For example: `v1.28.11`. See the compatibility matrix here: https://vcluster.com/docs/v0.19/deploying-vclusters/compat-matrix
+
+### 2.2. Ingress parameters
+
+Ingress parameters allow to connect app to the external world, by giving final user the URL to reach the app portal/entry point/API/etc.
+
+```json
+                "ingress_type": "subdomain",
+```
+
+`ingress_type` specify the type of ingress to be use. By default it is `subdomain`, which means that for each job, the ingress URL will be of format kns-job-XXX.mycluster.example.
+On some clusters, a subdomain based ingress cannot be used, and so instead of providing a subdomain based ingress, the url will be of format mycluster.example/XXX. To do so, set `ingress_type` to `prefix`. Note that not all KNS applications are compatible with this workaround (Kubeflow for example will not work).
+
+```json
+                "targets": [
+                    {
+                        "service_name": "web",
+                        "service_port": 8080,
+                        "service_namespace": "default"
+                    }
+                ],
+```
+
+`targets` list is then used to defind ingresses to spawn.
+For now, KNS will only consider the first item in the list. This limitation will be removed in later versions.
+
+The following parameters are to be set for each ingress target:
+
+* `service_name`: name of the service to link ingress to.
+* `service_port`: port of the service to link ingress to.
+* `service_namespace`: namespace where the service to link ingress to is.
+
+### 2.3. Templates parameters
+
+#### 2.3.1. Global settings
+
+```json
+                "templates_settings": {
+                    "global_retry": 20,
+                    "global_retry_sleep": 15
+                },
+```
+
+Templates are provided as a list, named `templates`, which will be executed in an ordered manner.
+
+While each item of the list can set its own parameters, the following global parameters are available, which apply to all items of the tempate list:
+
+* `global_retry`: how much retry to do if executing the item failed. For example, if the kubectl apply of the template fail, it will be retried. This is useful when a step needs another previous one to have completed to succeed (resources running).
```

Comment:
```suggestion
* `global_retry`: how much retry to do if executing the item failed. For example, if the kubectl apply of the template fail, it will be retried. This is useful when a step needs another previous one to have completed to succeed (resources running).
```
```suggestion
* `global_retry`: how much retry to do if executing the item failed. For example, if the `kubectl apply` of the template fails, it will be retried. This is useful when a step needs another previous one to be completed to succeed (resources running).
```